### ISSUE TITLE
Single-Plot only generates few points

### ISSUE THREAD

**Issue by memoryleak47** [hz-b/RAY-X#113]
The file given here:
https://github.com/hz-b/RAY-X/issues/94
only generates a two points when ran with -p

**Comment by memoryleak47** [comment]
![image](https://user-images.githubusercontent.com/5725430/198578451-558ef65f-78a3-4050-bbf6-8694c62849eb.png)
that's how it looks for me.

**Comment by Atraxus** [comment]
I get only one point, as far as I can see (this is on a6d3e3d4, the single-plot branch doesn't output anything)

![Figure_1](https://user-images.githubusercontent.com/13185253/198966816-b60f39a0-a8d3-470e-a5d8-dc0c660b364e.png)

**Comment by OSMaxwell** [comment]
This might be a bug in the tracing : 
As we're emulating RAY-UI : We need to look for rays that crossed the longest length 1234 etc
Here's an output of test_multi.rml
Left : extraParam, Right: amount
```
[            Plotter.cpp:157] 0:5840
[            Plotter.cpp:157] 1:4594
[            Plotter.cpp:157] 2:53862
[            Plotter.cpp:157] 3:62308
[            Plotter.cpp:157] 4:53617
[            Plotter.cpp:157] 5:4420
[            Plotter.cpp:157] 6:1107039
[            Plotter.cpp:157] 54:1
[            Plotter.cpp:157] 61:49366
[            Plotter.cpp:157] 62:187286
[            Plotter.cpp:157] 63:234534
[            Plotter.cpp:157] 64:187468
[            Plotter.cpp:157] 65:49660
[            Plotter.cpp:157] 212:1
[            Plotter.cpp:157] 454:1
[            Plotter.cpp:157] 612:2
[            Plotter.cpp:157] 654:1
```

You can see that the ones that meet 6 (ImagePlane) with the longest path are not that much!!

**Comment by OSMaxwell** [comment]
Here's a trace when looking only for the Imageplane being last as in 6XXXXX is accepeted
![image](https://user-images.githubusercontent.com/48519746/200844476-d202b564-1f48-4f00-a981-2c7d4f05a33f.png)

However this doesn't look like RAY-UI
![image](https://user-images.githubusercontent.com/48519746/200844972-ccf4acc8-6cc5-4b17-bad5-f05af7a74a4d.png)

The fix also breaks David's beamline which used to be correct when emulating the "sequential tracing" in plots

**Comment by memoryleak47** [comment]
>  As we're emulating RAY-UI : We need to look for rays that crossed the longest length 1234 etc

Can you explain what you mean with that?
I'm under the impression "single-plot" should render all rays ending in the ImagePlane, not only those taking the longest path to it, right?

**Comment by OSMaxwell** [comment]
Fixed in f48389a921b4ce6a2b7b92e3b71537e5307893aa . Use LONGEST_PATH if you want to revert back to old footprints
